src/lib/pmd/PmdCatalogWrapper.ts

```diff
@@ -86,7 +87,7 @@ export class PmdCatalogWrapper extends PmdSupport {
 
   protected buildCommand(): string {
     // TODO: We'll need to make sure this works on Windows.
-    return `java -cp "${this.buildClasspath().join(':')}" ${MAIN_CLASS} ${PMD_LIB} ${PMD_VERSION} ${SUPPORTED_LANGUAGES.join(',')}`;
+    return `java -cp "${this.buildClasspath().join(':')}" ${MAIN_CLASS} ${PMD_LIB} ${PMD_VERSION} ${SUPPORTED_LANGUAGES.join(',')} ${CUSTOM_RULE_REGISTER}`;
```

Comment:
You should be reading the register here, not passing its location into the Java. That's because the JARs and resources mentioned in the register need to be added to the classpath.

Reply:
Passing language and path makes the argument tedious. Passing the file and parsing details later seems a lot cleaner. However, I'm reconsidering passing register vs passing path entries because PmdWrapper.ts requires me to break it down. Making changes to see what sits better. Will update soon.

Reply:
I recognize that it's tedious, but if you don't add the custom JARs to the classpath, which you're not currently doing, the cataloger will not work without some extremely aggressive refactoring. And additionally, it's also just easier to read the paths here than to try to read them in Java where JSONs aren't natively supported.

Reply:
Yeah, I see what you mean. XML reader can't reach XML in JAR files if they are not in the classpath. I'm in the process of rewriting a portion of this.